app/models/work/date_of_work.rb

```diff
@@ -7,7 +7,14 @@ class DateOfWork
 
     validates :start_qualifier, inclusion: { in: START_QUALIFIERS, allow_blank: true }
     validates :finish_qualifier, inclusion: { in: FINISH_QUALIFIERS, allow_blank: true }
-    validates_format_of :start, with: /\A\d{4}(-\d{2}(-\d{2})?)?\Z/, message: "must be of format YYYY[-MM-DD]"
+    validates_format_of :start, with: /\A\d{4}(-\d{2}(-\d{2})?)?\z/,
```

Comment:
Note: the marker for the end of a line is \z, not \Z, so I changed it this too.

Reply:
I think \z and \Z actually mean subtly different things, not sure which is correct, let me google... 

https://stackoverflow.com/questions/2707870/whats-the-difference-between-z-and-z-in-a-regular-expression-and-when-and-how

> \Z  The end of the input but for the final terminator, if any
> \z  The end of the input

Weird, not sure which of those is 'right', but `\z` seems likely, thanks!

Reply:
I now realize it makes zero difference in this case actually, but \z is theoretically more correct.

Reply:
I guess it could make a difference if a newline somehow erroneously made it into the string, and your correction is correct!